Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkFilter.cc

```diff
@@ -0,0 +1,194 @@
+public:
+  explicit AlCaHcalIsotrkFilter(edm::ParameterSet const&, const AlCaHcalIsoTrk::Counters* count);
+  ~AlCaHcalIsotrkFilter() override;
+
+  static std::unique_ptr<AlCaHcalIsoTrk::Counters> initializeGlobalCache(edm::ParameterSet const&) {
+    return std::make_unique<AlCaHcalIsoTrk::Counters>();
+  }
+
+  bool filter(edm::Event&, edm::EventSetup const&) override;
+  void endStream() override;
+  static void globalEndJob(const AlCaHcalIsoTrk::Counters* counters);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  void endRun(edm::Run const&, edm::EventSetup const&) override;
+
+  // ----------member data ---------------------------
+  unsigned int nRun_, nAll_, nGood_, nLow_, nHigh_;
+  const double pTrackLow_, pTrackHigh_;
+  const int prescaleLow_, prescaleHigh_;
+  const std::string labelIsoTkVar_, labelIsoTkEvtVar_;
+  edm::EDGetTokenT<HcalIsoTrkCalibVariablesCollection> tokIsoTrkVar_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+AlCaHcalIsotrkFilter::AlCaHcalIsotrkFilter(edm::ParameterSet const& iConfig, const AlCaHcalIsoTrk::Counters* count)
+    : nRun_(0),
+      nAll_(0),
+      nGood_(0),
+      nLow_(0),
+      nHigh_(0),
+      pTrackLow_(iConfig.getParameter<double>("momentumLow")),
+      pTrackHigh_(iConfig.getParameter<double>("momentumHigh")),
+      prescaleLow_(iConfig.getParameter<int>("prescaleLow")),
+      prescaleHigh_(iConfig.getParameter<int>("prescaleHigh")),
+      labelIsoTkVar_(iConfig.getParameter<std::string>("isoTrackLabel")) {
+  // define tokens for access
+  tokIsoTrkVar_ = consumes<HcalIsoTrkCalibVariablesCollection>(labelIsoTkVar_);
+
+  edm::LogVerbatim("HcalIsoTrack") << "Parameters read from config file \n\t momentumLow_ " << pTrackLow_
+                                   << "\t prescaleLow_ " << prescaleLow_ << "\t momentumHigh_ " << pTrackHigh_
+                                   << "\t prescaleHigh_ " << prescaleHigh_ << "\n\t Labels " << labelIsoTkVar_;
+}  // AlCaHcalIsotrkFilter::AlCaHcalIsotrkFilter  constructor
+
+AlCaHcalIsotrkFilter::~AlCaHcalIsotrkFilter() {}
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool AlCaHcalIsotrkFilter::filter(edm::Event& iEvent, edm::EventSetup const& iSetup) {
+  bool accept(false);
+  ++nAll_;
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HcalIsoTrack") << "AlCaHcalIsotrkFilter::Run " << iEvent.id().run() << " Event "
+                                   << iEvent.id().event() << " Luminosity " << iEvent.luminosityBlock() << " Bunch "
+                                   << iEvent.bunchCrossing();
+#endif
+
+  edm::Handle<HcalIsoTrkCalibVariablesCollection> isotrkCalibColl;
+  iEvent.getByToken(tokIsoTrkVar_, isotrkCalibColl);
```

Comment:
This is discouraged, as you know, please do it with the new method, as you have done all the mitigations
```suggestion
  auto const& isotrkCalibColl = iEvent.get(tokIsoTrkVar_);
```